pkg/provision/primitives/counters.go

```diff
@@ -184,6 +186,35 @@ type resourceUnits struct {
 	CRU uint64 `json:"cru,omitempty"`
 }
 
+// CheckMemoryRequirements checks memory requirements for a reservation and compares it to whats in the counters
+// and what is the total memory on this node
+func (c *Counters) CheckMemoryRequirements(r *provision.Reservation, memStats *mem.VirtualMemoryStat) error {
+	switch r.Type {
+	case ContainerReservation:
+		requestedUnits, err := processContainer(r)
+		if err != nil {
+			return err
+		}
+
+		// If current MRU + requested MRU exceeds total, return an error
+		if uint64(c.CurrentUnits().Mru)+requestedUnits.HRU > memStats.Total {
+			return errors.New("reservation requested memory is to high")
```

Comment:
This is a misleading error message, it means that this memory size is not supported by the system while in the matter of fact it's only a node capacity issue. May be change the message to `not enough free resources to support this memory size' or something similar